src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -653,7 +653,7 @@ internal void InitializeForReflectionSerializer()
 
                 case JsonSerializerContext ctx when AppContextSwitchHelper.IsSourceGenReflectionFallbackEnabled:
                     // .NET 6 compatibility mode: enable fallback to reflection metadata for JsonSerializerContext
-                    _effectiveJsonTypeInfoResolver = JsonTypeInfoResolver.Combine(ctx, defaultResolver);
+                    _typeInfoResolver = JsonTypeInfoResolver.Combine(ctx, defaultResolver);
```

Comment:
`_typeInfoResolver` is the backing field for the public `TypeInfoResolver` property and is consulted by the copy constructor. The fallback is intentionally stored in a separate field so that any triggering of reflection fallback in the current options instance does not alter the behavior of either of these APIs.

Reply:
Thanks, I suspected this. I'll take a moment to figure out if there should be a test case that should have failed with this change.

Reply:
this is hard to validate - you'll lose fast path if you remove effective resolver

Reply:
Looks like we can close this PR?